### ISSUE TITLE
Pip internal API changes

### ISSUE THREAD

**Issue by chmue** [quanteda/spacyr#132]
The pip internal API you are using changed in pip 18.1 (see [here](https://github.com/naiquevin/pipdeptree/commit/ba67474e41aacfacd3811cb1f0dd5a16dc6251c8)).

```r
> library("spacyr")
> spacy_initialize("en-core-web-sm", virtualenv = "~/src/venv")
Error in py_run_file_impl(file, local, convert) : 
  ImportError: cannot import name 'get_installed_distributions' from 'pip' (/home/cmueller/src/venv/lib/python3.7/site-packages/pip/__init__.py)
```

It would probably make sense to switch this to a more stable solution as you are using an [internal API](https://mail.python.org/pipermail/distutils-sig/2017-October/031642.html) that is subject to change without notice.

**Comment by chmue** [comment]
There seems to be a solution in `setuptools` mentioned here https://github.com/pypa/pip/issues/5243#issuecomment-381513000

**Comment by chmue** [comment]
I provided a simple fix for the current problem in #133.

**Comment by chmue** [comment]
If you just need to detect the version of spacy which is loaded, you could use:
```python
import spacy
spacy.about.__version__
```

**Comment by chmue** [comment]
If you want to also compare the version numbers, you can use version number parsers/comparators provided by `distutils.version`, which is part of the standard library and, as far as I could see, available in all currently supported versions of Python.

```python
import spacy
from distutils.version import StrictVersion, LooseVersion
LooseVersion(spacy.about.__version__) >= LooseVersion("2.0.0")
```
`StrictVersion` requires more or less a version of SemVer so it might not support development versions. As long as you only need to check for the major version, `LooseVersion` should work and is more permissive.

**Comment by chmue** [comment]
Another solution would be to use https://packaging.pypa.io/en/latest/version/#packaging.version.parse

This is a better parser (that complies to the current PEP440 concerning version numbers). It is not part of the python standard library though, so not _guaranteed_ to be installed.

At the same time, it is a dependency of `setuptools` which in turn is a dependency of `pip` so it will work whenever your current approach with the pip internals works. There might be some edge cases where `pip`/`setuptools` is not available though. I suspect that this might be one of the problems in #124.

**Comment by chmue** [comment]
Fixed in #133
